src/test/resources/regressions/examples/spec_errors/visitor_pattern.gobra

```diff
@@ -148,8 +137,6 @@ pure func (self *evaluator) visitAddition(a *addition, ghost u int) int {
 
 
 (*evaluator) implements visitor {
-  pure (self *evaluator) visitVariable(v *variable, ghost u int) int { return self.visitVariable(v, u) }
-  pure (self *evaluator) visitConstant(c *constant, ghost u int) int { return self.visitConstant(c, u) }
   //:: ExpectedOutput(precondition_error)
   pure (self *evaluator) visitAddition(c *addition, ghost u int) int { return self.visitAddition(c, u) }
```

Comment:
Does this new approach support partial implementation proofs? E.g. if the implementation of `visitAddition` was correct, would the current proof inference generate proofs for the methods `visitVariable` and `visitConstant` and use the proof you provided for `visitAddition`?

Reply:
This is not a new approach. I just added that the desugarer actually generates the proof obligations. Yes, it can generated partial proofs.